lib/matplotlib/tests/test_legend.py

```diff
@@ -1667,3 +1667,78 @@ def test_boxplot_legend_labels():
     bp4 = axs[3].boxplot(data, label='box A')
     assert bp4['medians'][0].get_label() == 'box A'
     assert all(x.get_label().startswith("_") for x in bp4['medians'][1:])
+
+
+def test_patchcollection_legend():
+    # Test that PatchCollection labels show up in legend (issue #23998)
+    fig, ax = plt.subplots()
+
+    pc = mcollections.PatchCollection(
+        [
+            mpatches.Polygon([[0, 0], [100, 100], [200, 0]]),
+            mpatches.Polygon([[400, 0], [500, 100], [600, 0]]),
+        ],
+        label="patch collection", facecolor='blue', edgecolor='black'
+    )
+    ax.add_collection(pc, autolim=True)
+    ax.autoscale_view()
+
+    leg = ax.legend()
+
+    # Check that the legend contains our label
+    assert len(leg.get_texts()) == 1
+    assert leg.get_texts()[0].get_text() == "patch collection"
+
+    # Check that the legend handle exists and has correct colors
+    assert len(leg.legend_handles) == 1
+    legend_patch = leg.legend_handles[0]
+    assert_allclose(legend_patch.get_facecolor(),
+                    pc.get_facecolor()[0], rtol=1e-5)
+    assert_allclose(legend_patch.get_edgecolor(),
+                    pc.get_edgecolor()[0], rtol=1e-5)
+
+
+def test_patchcollection_legend_properties():
```

Comment:
You can combine `test_patchcollection_legend:` and `test_patchcollection_legend_properties():` b/c text is just another property

Reply:
ok